server/src/main/java/org/opensearch/indices/replication/copy/ReplicationCollection.java

```diff
@@ -221,29 +221,15 @@ public boolean cancelRecoveriesForShard(ShardId shardId, String reason) {
      * causes {@link ReplicationTarget#decRef()} to be called. This makes sure that the underlying resources
      * will not be freed until {@link ReplicationRef#close()} is called.
      */
-    public static class ReplicationRef implements AutoCloseable {
-
-        private final ReplicationTarget target;
-        private final AtomicBoolean closed = new AtomicBoolean(false);
+    public static class ReplicationRef extends GatedAutoCloseable<ReplicationTarget> {
```

Comment:
This class is now virtually identical to `RecoveryRef`, so IMO the next logical step is to absorb the common functionality into `GatedAutoCloseable` (and probably rename it to something more appropriate). At that point, both the `*Ref` classes can be deleted.

I'll work on this as a follow up.

Reply:
Refactored to `AutoCloseableRefCounted` in https://github.com/opensearch-project/OpenSearch/commit/74e04b55542b380209080ab6102b325869dcc017